src/jgikbase/idmapping/userhandlers/kbase_user_handler.py

```diff
@@ -0,0 +1,71 @@
+"""
+A ID mapper service user handler for KBase (https://kbase.us) user accounts.
+"""
+from jgikbase.idmapping.core.user_handler import UserHandler
+from jgikbase.idmapping.core.util import not_none
+from jgikbase.idmapping.core.user import AuthsourceID, User, Username
+from jgikbase.idmapping.core.tokens import Token
+import requests
+
+
+# WARNING - this is tested by mocking the requests library. The test suite never tests it against
+# an actual KBase auth server. If you make changes, you must manually test that everything works
+# against a live server.
+
+class KBaseUserHandler(UserHandler):
+    """
+    A user handler for the ID mapping service.
+
+    :ivar auth_url: The KBase authentication service url.
+    """
+
+    # TODO NOW caching
+
+    _KBASE = AuthsourceID('kbase')
+
+    def __init__(self, kbase_auth_url: str, kbase_token: Token) -> None:
+        '''
+        Create the handler.
+
+        :param kbase_auth_url: The url for the KBase authentication service.
+        :param kbase_token: A valid KBase user token. This is used for check the validity of
+            user names.
+        '''
+        not_none(kbase_auth_url, 'kbase_auth_url')
+        not_none(kbase_token, 'kbase_token')
+        if not kbase_auth_url.endswith('/'):
+            kbase_auth_url += '/'
+        self.auth_url = kbase_auth_url
+        self._token = kbase_token
+
+    def get_authsource_id(self) -> AuthsourceID:
+        return self._KBASE
+
+    def _check_error(self, r):
+        if r.status_code != 200:
+            try:
+                j = r.json()
+            except Exception as e:  # @UnusedVariable
+                raise IOError('Non-JSON response from KBase auth server, status code: ' +
```

Comment:
Maybe you could print the response text here for debugging -- I've seen some unknown errors from kbase services that are a bit hard to debug because nothing is printed

Reply:
I don't mean print exactly, but rather put the response text in the error message string.

Reply:
The problem is that if the client gets text back, it's probably pointed at god knows what endpoint, and so the returned text could be huge. I've added a TODO to log the returned page when it's not parseable as JSON instead, will that do for now? I haven't quite figured out how I'm going to handle the logging system yet, so I don't want to implement anything ATM.

Reply:
Logging sounds good